### PR TITLE
Update to 3.9, add framework, update tests

### PR DESCRIPTION
**Pull request by tlambert03**
This is a general update to testing strategies, and setup.py metadata

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! Maybe @ziyangczi wants to look too, something I was recently talking about with him was what it would be like to make something that can update existing cookiecutters too (like my napari-animation repo etc)

**Comment by tlambert03** [comment]
> make something that can update existing cookiecutters

yep, that would be awesome... but extremely hard to do with code that the user has likely modified in the interim.  You can have a look through [this discussion](https://github.com/cookiecutter/cookiecutter/issues/784) ... see also https://github.com/cruft/cruft, which purports to help with updating based on a template

**Comment by sofroniewn** [comment]
ooo, those links are very helpful! I wasn't familiar with cruft, that could be very helpful! I was particularly concerned with keeping the CI/ testing/ packaging stuff up to date which seems like its focus. Curious what you think @ziyangczi?

File: napari-{{cookiecutter.plugin_name}}/setup.py [hunk 1]
```diff
@@ -44,14 +44,13 @@ def read(fname):
     classifiers=[
         'Development Status :: 4 - Beta',
         'Intended Audience :: Developers',
+        "Framework :: napari"
```
Comment (jni) [inline napari-{{cookiecutter.plugin_name}}/setup.py:47]: Erm, there is a comma missing... ;) Not to mention double quotes 🤮 😜

Comment (tlambert03) [reply napari-{{cookiecutter.plugin_name}}/setup.py:47]: thanks will update.  but aren't the double quotes the standard?

Comment (tlambert03) [reply napari-{{cookiecutter.plugin_name}}/setup.py:47]: https://pypi.org/classifiers/

Comment (tlambert03) [reply napari-{{cookiecutter.plugin_name}}/setup.py:47]: ☝️ ignore this :) I'm an idiot (as discussed on zulip).  my brain went to "double colons" for some reason :)

**Comment by ziyangczi** [comment]
> ooo, those links are very helpful! I wasn't familiar with cruft, that could be very helpful! I was particularly concerned with keeping the CI/ testing/ packaging stuff up to date which seems like its focus. Curious what you think @ziyangczi? cc @justinelarsen

I have been thinking about the python compatibility issue if a plugin remains on an older version of python. I don't think this will make it to the original version of implementation focus, but I will add it to the considerations

**Comment by tlambert03** [comment]
A good point.  I do think we need to also keep in mind the possibility that a plugin may not _only_ be used in napari (like ome-zarr... which has a ton of other uses).  So even though this cookiecutter _starts_ as a napari plugin, whatever we do can't simply assume that that is their only goal (and must allow them to make their own choices like python support)

**Pull request merged**